graylog2-server/src/main/java/org/graylog2/indexer/ranges/IndexRange.java

```diff
@@ -16,18 +16,38 @@
  */
 package org.graylog2.indexer.ranges;
 
-import org.graylog2.plugin.database.Persisted;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
 import org.joda.time.DateTime;
 
-/**
- * @author Dennis Oelkers <dennis@torch.sh>
- */
-public interface IndexRange extends Persisted {
-    String getIndexName();
+import java.util.Comparator;
+
+@AutoValue
+@JsonAutoDetect
+public abstract class IndexRange {
+    public static final Comparator<IndexRange> COMPARATOR = new IndexRangeComparator();
+
+    @JsonProperty
+    public abstract String indexName();
+
+    @JsonProperty
+    public abstract DateTime begin();
```

Comment:
"start" would be the same as "end" (it was misnamed from the start – pun intended) and I think that would be much more irritating than jus changing the format.